### PR TITLE
Move OAuth mediator from carbon-mediation

### PR DESCRIPTION
**Pull request by Vathsan**
move org.wso2.carbon.identity.oauth.mediator to org.wso2.micro.integrator.mediator.oauth

### CODE REVIEW

File: components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/pom.xml [hunk 1]
```diff
@@ -0,0 +1,93 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+ ~ Copyright (c) 2009-2011, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (sdkottegoda) [inline components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/pom.xml]: Shall we update copyrights?

File: components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/MediatorException.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+package org.wso2.micro.integrator.mediator.oauth;
+
+/**
+ * Exception for the sequence editor mediator service.
+ */
+@SuppressWarnings({"UnusedDeclaration"})
+public class MediatorException extends RuntimeException {
+
+    public MediatorException() {
```
Comment (sdkottegoda) [inline components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/MediatorException.java]: Do we need this constructor?

File: components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/OAuth2TokenValidationServiceClient.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+public class OAuth2TokenValidationServiceClient {
+
+    private OAuth2TokenValidationServiceStub stub = null;
+    private static final Log log = LogFactory.getLog(OAuth2TokenValidationServiceClient.class);
+
+    /**
+     * OAuth2TokenValidationService Admin Service Client
+     *
+     * @param backendServerURL
```
Comment (sdkottegoda) [inline components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/OAuth2TokenValidationServiceClient.java]: Shall we add descriptions for the input paramters?

File: components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/OAuth2TokenValidationServiceClient.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+     * @throws Exception
+     */
+    public OAuth2TokenValidationServiceClient(String backendServerURL, String username, String password,
+                                              ConfigurationContext configCtx) throws Exception {
+        String serviceURL = backendServerURL + "OAuth2TokenValidationService";
+        try {
+            stub = new OAuth2TokenValidationServiceStub(configCtx, serviceURL);
+            MicroCarbonUtils.setBasicAccessSecurityHeaders(username, password, true, stub._getServiceClient());
+        } catch (AxisFault e) {
+            log.error("Error initializing OAuth2 Client");
```
Comment (sdkottegoda) [inline components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/OAuth2TokenValidationServiceClient.java]: Let's not log and throw the exception. Throwing shoud be enough, WDYT?

File: components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/OAuth2TokenValidationServiceClient.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+            throw new Exception("Error initializing OAuth Client", e);
+        }
+    }
+
+    /**
+     * Validates the OAuth 2.0 request
+     *
+     * @param accessTokenIdentifier
```
Comment (sdkottegoda) [inline components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/OAuth2TokenValidationServiceClient.java]: Let's add descriptions to input parameter in all places?

File: components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/OAuth2TokenValidationServiceClient.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        OAuth2TokenValidationRequestDTO_OAuth2AccessToken accessToken =
+                new org.wso2.carbon.identity.oauth2.stub.dto.OAuth2TokenValidationRequestDTO_OAuth2AccessToken();
+        accessToken.setTokenType(OAuthConstants.BEARER_TOKEN_TYPE);
+        accessToken.setIdentifier(accessTokenIdentifier);
+        oauthReq.setAccessToken(accessToken);
+        oauthReq.setContext(params.toArray(new OAuth2TokenValidationRequestDTO_TokenValidationContextParam[params.size()]));
+        try {
+            return stub.validate(oauthReq);
+        } catch (RemoteException e) {
+            log.error("Error while validating OAuth2 request");
```
Comment (sdkottegoda) [inline components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/OAuth2TokenValidationServiceClient.java]: Let's not log and throw. Let's correct in all places

File: components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/OAuthConstants.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    public static final String OAUTH_NONCE = "oauth_nonce";
+    public static final String OAUTH_TIMESTAMP = "oauth_timestamp";
+    public static final String OAUTH_CONSUMER_KEY = "oauth_consumer_key";
+    public static final String OAUTH_CALLBACK = "oauth_callback";
+    public static final String OAUTH_SIGNATURE_METHOD = "oauth_signature_method";
+    public static final String OAUTH_SIGNATURE = "oauth_signature";
+    public static final String SCOPE = "scope";
+    public static final String OAUTH_DISPLAY_NAME = "xoauth_displayname";
+
+
```
Comment (sdkottegoda) [inline components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/OAuthConstants.java]: Shall we remove unnecessary new lines?

File: components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/OAuthMediator.java [hunk 8]
```diff
@@ -0,0 +1,423 @@
...
+    public final static String DEFAULT_AXIS2_XML = "./samples/axis2Client/client_repo/conf/axis2.xml";
+
+    /**
+     * {@inheritDoc}
+     */
+    public void init(SynapseEnvironment synEnv) {
+        try {
+            cfgCtx =
+                    ConfigurationContextFactory.createConfigurationContextFromFileSystem(clientRepository != null
+                                    ? clientRepository
```
Comment (sdkottegoda) [inline components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/OAuthMediator.java]: formatting issue?

File: components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/config/xml/OAuthMediatorFactory.java [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+import org.apache.synapse.config.xml.XMLConfigConstants;
+import org.wso2.micro.integrator.mediator.oauth.OAuthMediator;
+import org.wso2.micro.integrator.mediator.oauth.MediatorException;
+
+import javax.xml.namespace.QName;
+import java.util.Properties;
+
+/**
+ * <oauthService remoteServiceUrl = "https://identityserever/services/OAuthService"
+ * />
```
Comment (sdkottegoda) [inline components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/config/xml/OAuthMediatorFactory.java]: Shall we merge this to the previous line?

File: components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/pom.xml (hunk 1 above)
Comment (Vathsan) [reply components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/pom.xml]: Done

File: components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/MediatorException.java (hunk 2 above)
Comment (Vathsan) [reply components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/MediatorException.java]: Removed MediarException class. Refactored oauth mediator to throw SynapseException in 563f10f6e2d2e0936ec3843f9ae35d84a369bef9

File: components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/OAuth2TokenValidationServiceClient.java (hunk 3 above)
Comment (Vathsan) [reply components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/OAuth2TokenValidationServiceClient.java]: Done in 563f10f6e2d2e0936ec3843f9ae35d84a369bef9

File: components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/OAuth2TokenValidationServiceClient.java (hunk 4 above)
Comment (Vathsan) [reply components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/OAuth2TokenValidationServiceClient.java]: Fixed in 563f10f6e2d2e0936ec3843f9ae35d84a369bef9

File: components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/OAuth2TokenValidationServiceClient.java (hunk 5 above)
Comment (Vathsan) [reply components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/OAuth2TokenValidationServiceClient.java]: Done in 563f10f6e2d2e0936ec3843f9ae35d84a369bef9

File: components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/OAuth2TokenValidationServiceClient.java (hunk 6 above)
Comment (Vathsan) [reply components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/OAuth2TokenValidationServiceClient.java]: Fixed in 563f10f6e2d2e0936ec3843f9ae35d84a369bef9

File: components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/OAuthConstants.java (hunk 7 above)
Comment (Vathsan) [reply components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/OAuthConstants.java]: Done

File: components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/OAuthMediator.java (hunk 8 above)
Comment (Vathsan) [reply components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/OAuthMediator.java]: Done

File: components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/config/xml/OAuthMediatorFactory.java (hunk 9 above)
Comment (Vathsan) [reply components/mediation/mediators/oauth-mediator/org.wso2.micro.integrator.mediator.oauth/src/main/java/org/wso2/micro/integrator/mediator/oauth/config/xml/OAuthMediatorFactory.java]: Done in 563f10f6e2d2e0936ec3843f9ae35d84a369bef9

**Pull request merged**